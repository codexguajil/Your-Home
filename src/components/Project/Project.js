import React from 'react';
import { Link } from 'react-router-dom';
import homeIcon from '../../assets/home.png';
import DeleteConfirm from '../../components/DeleteConfirm/DeleteConfirm';

const Project = (props) => {
  const { id, name, address, description, rooms } = props;
  return (
    <div className='container'>
      <Link to={`/projects/${id}`}>
        <div className='project'>
          <div className='project-head'>
            <img src={homeIcon} className='home' alt='home-icon'/>
            <h2>{name}</h2>
          </div>
          <div className='project-description'>
            <p>{address}</p>
            <p>{description}</p>
            <p>{rooms.length} Rooms</p>
          </div>
        </div>
      </Link>
      <div className='project-delete'>
        <DeleteConfirm type="project"/>
      </div>
    </div>
  );
}

export default Project;