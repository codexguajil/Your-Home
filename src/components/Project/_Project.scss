.project {
  @include card-layout;
  height: 300px;
  width: 300px;
  text-align: center;
  opacity: 0.7;
  cursor: pointer;
  position: relative;
  &:hover {
    opacity: 1;
  }
}

.project-head {
  display: flex;
  justify-content: space-between;
  height: 80px;

  .home {
    height: 50px;
    width: 50px;
    // position: absolute;
    // left: 15px;
    // top: 15px;
  }
  .fa-trash-alt {
    font-size: 40px;
    margin-top: 5px;
    opacity: .1;
  }
}

