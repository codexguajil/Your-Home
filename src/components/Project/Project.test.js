import React from 'react';
import { shallow } from 'enzyme';
import Project from './Project';

describe('Project', () => {
  let wrapper;
  let mockRooms;
  beforeEach(() => {
    mockRooms =  [
          {
              "name": "Living Room 1",
              "type": "Living Room",
              "description": "Northeast living room",
              "roomMaterials": [{}]
          },
          {
              "name": "Room 2",
              "type": "Kitchen",
              "description": "Big Kitchen",
              "roomMaterials": [{}]
          }
      ]
    wrapper = shallow(<Project rooms={mockRooms}/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});