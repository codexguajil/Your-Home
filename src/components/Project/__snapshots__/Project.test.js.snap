// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match snapshot 1`] = `
<div
  className="container"
>
  <Link
    to="/projects/undefined"
  >
    <div
      className="project"
    >
      <div
        className="project-head"
      >
        <img
          alt="home-icon"
          className="home"
          src="home.png"
        />
        <h2 />
      </div>
      <div
        className="project-description"
      >
        <p />
        <p />
      </div>
    </div>
  </Link>
  <div
    className="project-delete"
  >
    <ConnectFunction
      type="project"
    />
  </div>
</div>
`;
